### PR TITLE
chore: fixed license URL in pom.xml

### LINKED ISSUE #3891: License link is broken
The projects pom specifies as a license link: http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.txt
This page is, however not reachable. (timeout)
The correct value seems to be http://cecill.info/licences/Licence_CeCILL-C_V1-en.txt (without `www`).

### PR DESCRIPTION
**Pull request by jakobbraun**
The current URL is not reachable (timeout).
Closes #3891

### CODE REVIEW

**Comment by monperrus** [comment]
Thanks for the update. Spoon has been dual-licensed MIT / Ceccil-C for some time. Would you be able to add the MIT block?

**Comment by jakobbraun** [comment]
sure. I updated the PR.

File: pom.xml [hunk 1]
```diff
@@ -27,7 +27,12 @@
       <license>
           <name>CeCILL-C</name>
           <comments>French equivalent to LGPL</comments>
-          <url>http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.txt</url>
+          <url>http://cecill.info/licences/Licence_CeCILL-C_V1-en.txt</url>
```
Comment (slarse) [inline pom.xml:30]: While we're at it, could we make this https? There's no auto-redirect from the site, but explicitly specifying the https protocol works.

```suggestion
          <url>https://cecill.info/licences/Licence_CeCILL-C_V1-en.txt</url>
```

Comment (jakobbraun) [reply pom.xml]: :+1:

Comment (slarse) [reply pom.xml]: Great!

**Review by slarse — APPROVED** [review]

**Comment by slarse** [comment]
@monperrus Note that you need to approve the GHA workflow run as @jakobbraun is a first-time contributor. It's a new security feature in GHA. Changes LGTM.

**Comment by monperrus** [comment]
Thanks a lot @jakobbraun

**Pull request merged**